### DISCUSSION TITLE
datagriff/2024/12/22/create-a-cracker-of-an-open-api-contract-with-vs-code-spectral-prism-and-schemathesis

### DISCUSSION THREAD

**Discussion by giscus** [hungovercoders-blog/datagriff#12] [Announcements]
# datagriff/2024/12/22/create-a-cracker-of-an-open-api-contract-with-vs-code-spectral-prism-and-schemathesis

Create a Cracker of an Open API Contract with VS Code, Spectral, Prism and Schemathesis

https://blog.hungovercoders.com/datagriff/2024/12/22/create-a-cracker-of-an-open-api-contract-with-vs-code-spectral-prism-and-schemathesis.html

**Comment by Stranger6667** [comment]
Hi! Schemathesis author here. Don't you mind if I include this article under the "Resources" section in the Schemathesis docs?

**Reply by dataGriff** [reply]
Please do. I'll be honest I'm not sure if I'm using schemathesis correctly at this point, but if it helps people think of the tool for different use cases, that can only be good!

**Reply by Stranger6667** [reply]
Thanks! I think I'd add `--checks all` so it runs all the checks. It is also something to adjust in Schemathesis itself - the "default" command without or with minimal configuration should yield as much value as possible. For this reason, I am reworking the CLI so it will enable `--checks all` by default and maybe a few more helpful options.

**Reply by dataGriff** [reply]
Do you think this was a redundant use of schemathesis? I'm using the contract as a mechanism to test the mock of the contract... Feels like it could never fail. I mean I'm glad it gives me confidence in the contract and I know consumers can mock safely, but only by being tested by it's own specification

**Comment by Stranger6667** [comment]
And thank you for the article! I have a lot of useful tools to check out :)
